18-explaining-models-and-predictions.Rmd

```diff
@@ -1,6 +1,313 @@
+
+:::rmdnote
+There are two types of model explanations, **global** and **local**. Global model explanations provide an overall understanding aggregated over a whole set of observations; local model explanations provide information about a prediction for a single observation.
+:::
+
+
+## Software for model explanations
+
+The tidymodels framework does not itself contain software for model explanations. Instead, models trained and evaluated with tidymodels can be explained with other, supplementary software in R packages such as `r pkg(lime)`, `r pkg(vip)`, and `r pkg(DALEX)`. We ourselves often choose: 
+
+- `r pkg(vip)` functions when we want to use _model-based_ methods that take advantage of model structure (and are often faster), and 
+- `r pkg(DALEX)` functions when we want to use _model-agnostic_ methods that can be applied to any model. 
+
+In Chapters \@ref(resampling) and \@ref(compare), we trained and compared several models to predict the price of homes in Ames, IA, including a linear model with interactions and a random forest model.
+
+```{r include=FALSE}
+bind_rows(
+  augment(lm_fit, ames_train) %>% mutate(model = "lm + interactions"),
+  augment(rf_fit, ames_train) %>% mutate(model = "random forest")
+) %>%
+  ggplot(aes(Sale_Price, .pred, color = model)) +
+  geom_abline(col = "gray50", lty = 2) + 
+  geom_point(alpha = 0.5, show.legend = FALSE) +
+  facet_wrap(vars(model))
+```
+
+Let's build model-agnostic explainers for both of these models to why they make the predictions they do. We can use the `r pkg(DALEXtra)` add-on package for `r pkg(DALEX)`, which provides support for tidymodels. @Biecek2021 provide a thorough exploration of how to use `r pkg(DALEX)` for model explanations; this chapter only summarizes some important approaches, specific to tidymodels. To compute any kind of model explanation, global or local, using `r pkg(DALEX)`, we first create an _explainer_ for each model.
+
+```{r message=FALSE}
+library(DALEXtra)
+vip_features <- c("Neighborhood", "Gr_Liv_Area", "Year_Built", 
+                  "Bldg_Type", "Latitude", "Longitude")
+
+vip_train <- ames_train %>% 
+  select(all_of(vip_features))
+
+explainer_lm <- explain_tidymodels(
+  lm_fit, 
+  data = vip_train, 
+  y = ames_train$Sale_Price,
+  label = "lm + interactions",
+  verbose = FALSE
+)
+
+explainer_rf <- explain_tidymodels(
+  rf_fit, 
+  data = vip_train, 
+  y = ames_train$Sale_Price,
+  label = "random forest",
+  verbose = FALSE
+)
+```
+
+
+:::rmdwarning
+A linear model is typically straightforward to interpret and explain; you may not often find yourself computing separate model explanations for a linear model. However, it can sometimes be difficult to understand or explain the predictions of even a linear model once it has splines and interaction terms!
+:::
+
```

Comment:
Can you talk up-front about what is being measures for importance? 

Maybe use terminology like: 

- _basic predictors_: the original data columns (perhaps with simple transformations such as the log, etc). 
- _derived features_: columns created in the data set form the original predictors that are calculated from one or more of the basic predictors (e.g. spline terms, PCA components, ratios)

I don't recall seeing existing terminology in the DALEX book (but maybe I missed it).